modules/neo4j/src/main/java/org/testcontainers/containers/Neo4jContainer.java

```diff
@@ -250,4 +257,9 @@ private static String formatConfigurationKey(String plainConfigKey) {
             .replaceAll("_", "__")
             .replaceAll("\\.", "_"));
     }
+
+    private boolean usesVersion(String version) {
```

Comment:
We're a bit cautious that simply looking at the start of the version/tag could be a bit fragile in the case of custom images that people may be using, or simply surprising version numbers from any source.

Please could you have a look at how a similar problem is handled in `LocalStackContainer`?
https://github.com/testcontainers/testcontainers-java/blob/master/modules/localstack/src/main/java/org/testcontainers/containers/localstack/LocalStackContainer.java#L100

There are three main facets which (I hope!) add up to a safer implementation:
1. use our ComparableVersion utility class to do a semver comparison, not startsWith
2. gracefully fall back if a custom or non-parsable version is found - clearly logging what’s happened and how to change it
3. provide a constructor to allow the automatic version detection to be overridden, if a custom version string is being used that clashes with the original

It's not perfect - edge cases can still exist - but this seems to be working well for users.

Reply:
Thanks for your feedback. I changed the code to use `ComparableVersion` (hopefully right).
I refrained from the additional constructor because the feature can only used with older versions of Neo4j already today.
I do not want to have
a) an additional constructor added to the mix of the already available ones. This could confuse the users.
b) encourage users to use an older version for convenience
Both might even result in users trying to somehow hack their way around the restriction and end up with a broken database inside the container.